app/routes.py

```diff
@@ -1 +1,271 @@
+    goal.description = request_body["description"]
+
+    db.session.commit()
+
+    return make_response(jsonify({
+        "goal": Goal.to_dict(goal)})), 200
+
+@bp.route("<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response(jsonify({
+        "details" : f"Task {task_id} \"{task.title}\" successfully deleted"}))
+
+
+@goal_bp.route("<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response(jsonify({
+        "details" : f"Goal {goal_id} \"{goal.title}\" successfully deleted"}))
+
+
+def use_slack_bot(task):
+    PATH = "https://slack.com/api/chat.postMessage"
+    SLACK_API_KEY = os.environ.get("API_KEY")
+    
+    query_params = {
+        "channel": "#task-notifications",
+        "text" : f"Someone just completed the task {task.title}"
+        }
+
+    requests.post(PATH, params=query_params, headers={
+        "Authorization": SLACK_API_KEY
+    })
+
+@bp.route("<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete_on_incompleted_task(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = datetime.now()
+    db.session.commit()
+    use_slack_bot(task)
+    return {"task":task.to_dict()}, 200
+
+    
+
+
+
+
+@bp.route("<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete_on_completed_task(task_id):
+    try:
+        task = validate_model(Task, task_id)
+        task.completed_at = None
+
+        db.session.commit()
+
+        return make_response(jsonify({
+            "task": Task.to_dict(task)}))
+    except:
+        abort(make_response({"message" : f"task {task_id} not found"}, 404))
+
+@bp.route("<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete_on_completed_task(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = datetime.utcnow()
+    task.is_complete = True
+    task_dict = task.to_dict()
+
+    db.session.commit()
+
+    return make_response(jsonify({
+        "task": task_dict}))
+
+@bp.route("<task_id>/mark_incomplete", methods=["PATCH"])
+
+def mark_incomplete_on_incompleted_task(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = None
+
+    db.session.commit()
+
+    return make_response(jsonify({
+        "task": Task.to_dict(task)}))
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def send_list_of_tasks_to_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    goal.tasks = []
+
+    for task_id in request_body["task_ids"]:
+        task = validate_model(Task, task_id)
+        goal.tasks.append(task)
+
+
+    # db.session.add()
+    db.session.commit()
+
+    return make_response(jsonify({
+        "id" : goal.goal_id,
+        "task_ids" : request_body["task_ids"]
```

Comment:
Instead of using `request_body["task_ids"]` from the client's request to echo back what we received, we could get the task ids from `goal` itself.

You could create a helper method in goal.py to grab the task_ids from an instance of goal:
```python
 def task_ids(self):
        return [task.id for task in self.tasks]
```